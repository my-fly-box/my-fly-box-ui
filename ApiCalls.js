export const fetchFlies = async () => {
  try {
    const response = await fetch(
      "https://my-fly-box-api.herokuapp.com/api/v1/flies"
    );
    const flyData = await response.json();
    return flyData;
  } catch (error) {
    console.error(error);
  }
};

export const addFlyToAPI = async (flyData) => {
  try {
    const response = await fetch('https://my-fly-box-api.herokuapp.com/api/v1/flies', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json;charset=utf-8'
      },
      body: JSON.stringify(flyData),
    });
    const flyDataResponse = await response.json();
    return flyDataResponse;
  } catch (error) {
    console.error(error);
  }
};

export const addUpdatedFly = async (flyData) => {
  try {
    const response = await fetch(`https://my-fly-box-api.herokuapp.com/api/v1/flies/${flyData.id}`, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json;charset=utf-8'
      },
      body: JSON.stringify(flyData.attributes),
    });
    const flyDataResponse = await response.json();
    return flyDataResponse;
  } catch (error) {
    console.error(error);
  }
}
export const deleteFly = async (flyId) => {
    try {
      const response = await fetch(
        `https://my-fly-box-api.herokuapp.com/api/v1/flies/${flyId}`,
        {
          method: "DELETE",
          headers: {
            "Content-Type": "application/json;charset=utf-8",
          },
        }
      );
      const deletionResponse = await response.json();
      console.log(deletionResponse);
    } catch (error) {
      console.error(error);
    }
  };

  export const addFishToAPI = async (fishData) => {
    try {
      const response = await fetch('https://my-fly-box-api.herokuapp.com/api/v1/catches', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json;charset=utf-8'
        },
        body: JSON.stringify(fishData),
      });
      const fishDataResponse = await response.json();
      return fishDataResponse;
    } catch (error) {
      console.error(error);
    }
  };
