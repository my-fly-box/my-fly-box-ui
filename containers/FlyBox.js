import React, { Component } from "react";
import { StyleSheet, Text, View, FlatList } from "react-native";
import Fly from "../components/Fly";
import { setFlies, removeFly } from "../actions";
import { connect } from "react-redux";
import { fetchFlies, deleteFly } from "../ApiCalls";

class FlyBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentFlies: [],
      isLoading: true,
    };
  }

  async componentDidMount() {
    fetchFlies()
      .then((data) => this.props.setFlies(data.data))
      .then((fly) =>
        this.setState({ currentFlies: fly.data, isLoading: false })
      );
  }

  handleDelete = (flyId) => {
    deleteFly(flyId);
    this.props.removeFly(flyId);
  };

  checkFlyBox = () => {
    if (this.props.currentFlies.length > 0 && !this.state.isLoading) {
      // sorting flies in alphabetical order:
      let flies = this.props.currentFlies.sort((a, b) => {
        let nameA = a.attributes.name.toUpperCase();
        let nameB = b.attributes.name.toUpperCase();
        return (nameA < nameB) ? -1 : (nameA > nameB) ? 1 : 0;
      })
      
      return (
        <FlatList
          style={styles.fly}
          data={flies}
          renderItem={({ item }) => (
            <Fly
              fly={item}
              navigation={this.props.navigation}
              handleDelete={this.handleDelete}
            />
          )}
        />
      );
    } else if (this.state.isLoading) {
      return (
        <View
          style={{
            alignItems: "center",
            justifyContent: "center",
            paddingTop: 50,
          }}
        >
          <Text style={{ fontSize: 30, fontWeight: "bold" }}>
            Loading Fly Data...
          </Text>
        </View>
      );
    } else {
      return (
        <View
          style={{
            alignItems: "center",
            justifyContent: "center",
            paddingTop: 50,
          }}
        >
          <Text style={{ fontSize: 30, fontWeight: "bold" }}>
            It doesn't look like you have any flies saved to your fly box.
          </Text>
        </View>
      );
    }
  };

  render() {
    return (
      <View style={styles.container}>
        {this.checkFlyBox()}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#f8f8ff",
    padding: 10,
    alignItems: "center",
  },
  fly: {
    height: 60,
    width: "98%",
  },
  headerFont: {
    fontSize: 18,
    fontWeight: "bold",
    color: "#2A9D8F",
  },
});

const mapStateToProps = (state) => ({
  currentFlies: state.currentFlies,
});

const mapDispatchToProps = (dispatch) => ({
  setFlies: data => dispatch( setFlies(data) ),
  removeFly: id => dispatch( removeFly(id) ),
});

export default connect(mapStateToProps, mapDispatchToProps)(FlyBox);
