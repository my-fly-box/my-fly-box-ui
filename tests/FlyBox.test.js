import React from 'react';
import 'react-native'
import FlyBox from '../containers/FlyBox'
import { Provider } from 'react-redux'
import rootReducer from '../reducers/index'
import { createStore } from 'redux'
import renderer from 'react-test-renderer'
import { fetchFlies } from '../ApiCalls'

jest.mock('../ApiCalls')


describe('FlyBox', () => {
  let mockFlyData
  beforeEach(() => {
    mockFlyData = [
      {
        "id": "18",
        "type": "fly",
        "attributes": {
            "id": 18,
            "name": "Copper John",
            "size": 16,
            "color": "Copper",
            "category": "Nymph",
            "amount": 3
        }
    }
    ]
  })

  it('renders the FlyBox', async () => {
    const testStore = createStore(rootReducer)
    const testWrapper = <Provider store={testStore}><FlyBox /></Provider>
    fetchFlies.mockResolvedValueOnce(mockFlyData)
    const snap = renderer.create(
      testWrapper
    ).toJSON();
    expect(snap).toMatchSnapshot();
  })
}) 