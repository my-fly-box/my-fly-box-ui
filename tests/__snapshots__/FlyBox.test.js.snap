// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlyBox renders the FlyBox 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f8f8ff",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingTop": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      Loading Fly Data...
    </Text>
  </View>
</View>
`;
